### PR TITLE
Improvement : typo fix in post encoder and replaced static hashmap with array.

### PR DESCRIPTION
**Pull request by doom369**
Motivation:

HttpPostRequestEncoder has typo.

Modification:

Typo fixed and map replaced with array to do not allocate iterator for RFC3986 encoder.

Result:

Fixed typo, no more iterator on interation.

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java [hunk 1]
```diff
@@ -92,12 +92,14 @@
...
     }
 
-    private static final Map<Pattern, String> percentEncodings = new HashMap<Pattern, String>();
+    private static final SimpleImmutableEntry[] percentEncodings;
 
     static {
-        percentEncodings.put(Pattern.compile("\\*"), "%2A");
-        percentEncodings.put(Pattern.compile("\\+"), "%20");
-        percentEncodings.put(Pattern.compile("%7E"), "~");
+        percentEncodings = new AbstractMap.SimpleImmutableEntry[] {
```
Comment (netkins) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java]: MAJOR Unnecessary use of fully qualified name 'AbstractMap.SimpleImmutableEntry' due to existing static import 'java.util.AbstractMap.SimpleImmutableEntry' [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)

Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java]: @doom369 - just correct this issue then lgtm

**Comment by Scottmitch** [comment]
@doom369 - Can you also sign the [ICLA](https://docs.google.com/spreadsheet/viewform?formkey=dHBjc1YzdWhsZERUQnhlSklsbG1KT1E6MQ)?

**Comment by doom369** [comment]
@Scottmitch fixed.

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java [hunk 2]
```diff
@@ -92,12 +92,14 @@
         HTML5
     }
 
-    private static final Map<Pattern, String> percentEncodings = new HashMap<Pattern, String>();
+    private static final SimpleImmutableEntry[] percentEncodings;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java]: @doom369 I think you can use `Entry[] percentEncodings`. Always better to use the interface and not the actually impl when possible.

**Comment by doom369** [comment]
@normanmaurer agree. fixed.

**Comment by normanmaurer** [comment]
@Scottmitch please merge once you are satisfied.

**Comment by Scottmitch** [comment]
4.1 (2d11331)

**Comment by Scottmitch** [comment]
thanks @doom369 !

**Pull request closed without merging**